### ISSUE TITLE
Push channel JSON over binary

### ISSUE THREAD

**Issue by josevalim** [phoenixframework/phoenix#5815]
Both Bandit and Cowboy validate UTF-8 of text frames, per spec, but Channels push JSON which are also validated against UTF-8. Therefore, I suggest we use a binary frame for pushing JSON.

Our serializers already handle binary frames, but they always start with the bytes 0, 1, and 2. So there is no clash with JSON and we can start sending JSON over binaries frames without introducing a new serializer version. However, we may want to have a flag on the client that forces it over text instead for those who for some reason need to run against other serializers.

/cc @mtrudel

**Comment by mtrudel** [comment]
Good catch! As you know, UTF-8 validation is a surprisingly expensive operation and is usually near the top of profiling analysis of Bandit's WebSocket stack. It's an expensive thing to double up on for sure.

I agree that binary probably makes the most sense here (the spec is pretty loose about which one to use where; binary is mostly considered a simpler 'all bets are off' option that would apply well here). Absent that, both [Bandit](https://hexdocs.pm/bandit/Bandit.html#t:websocket_options/0) and [Cowboy](https://ninenines.eu/docs/en/cowboy/2.10/manual/cowboy_websocket/) have flags to disable UTF-8 validation.

**Comment by SteffenDE** [comment]
![Image](https://github.com/user-attachments/assets/5cd3980d-ce36-406d-8cd5-e3cfcaa9b44e)

One downside is that it's much harder to see what's going on.

> and we can start sending JSON over binaries frames without introducing a new serializer version

But how would this make sure that existing clients are compatible?

**Comment by josevalim** [comment]
> But how would this make sure that existing clients are compatible?

The clients would have to be updated for sure but it is a smaller change than introducing a new mechanism.

**Comment by josevalim** [comment]
Although perhaps the best option is to better document disabling the frame validation to be disabled if folks want to squeeze extra performance.

**Comment by SteffenDE** [comment]
Should we disable it by default and put a comment in the code saying that it should not be disabled if using raw websockets?
